File: Ruby/wordy.rb
class WordProblem
  def initialize(phrase)
    phrase = phrase.slice!(8..-2)
    phrase = phrase.gsub(' by', ' ')
    @phrase = phrase.split(' ')  
    # puts @phrase
  end

  def answer
    if @phrase.length == 2 then
      raise ArgumentError
    end
    if not Integer(@phrase[0]) then
      raise ArgumentError
    end
    case @phrase[1]
    when "plus"
      result = @phrase[0].to_i + @phrase[2].to_i 
    when "divided"
      result = @phrase[0].to_i / @phrase[2].to_i
    when "multiplied"
      result = @phrase[0].to_i * @phrase[2].to_i
    when "minus"
      result = @phrase[0].to_i - @phrase[2].to_i
    end
    if @phrase.length == 5 then
      case @phrase[3]
      when "plus"
        result = result + @phrase[4].to_i 
      when "divided"
        result = result / @phrase[4].to_i
      when "multiplied"
        result = result * @phrase[4].to_i
      when "minus"
        result = result - @phrase[4].to_i
      end
    end
    return result
  end
end
    
  problem = WordProblem.new("What is 1 plus 1 plus 1?")
  puts problem.answer

  problem = WordProblem.new("What is 10 divided by 2 plus 1?")
  puts problem.answer

  problem = WordProblem.new("What is 1 minus 1 plus 1?")
  puts problem.answer

File: Python/wordy.py
def answer(question):
    equation = question[8:-1]
    equation = equation.replace(" by ", " ")
    split_equation = equation.split()
    print(split_equation)

###    Numbers    ###
    
    if len(split_equation) == 1:
        return(int(split_equation[0]))
        
###    Single Operations    ###

    elif len(split_equation) == 3:
        if split_equation[1] == "plus":
            return(int(split_equation[0]) + int(split_equation[2]))
        elif split_equation[1] == "minus":
            return(int(split_equation[0]) - int(split_equation[2]))
        elif split_equation[1] == "multiplied":
            return(int(split_equation[0]) * int(split_equation[2]))
        elif split_equation[1] == "divided":
            return(int(split_equation[0]) / int(split_equation[2]))
        elif split_equation[0] == "plus" or split_equation[2] == "plus":
            raise ValueError("syntax error")
        else:
            raise ValueError("unknown operation")

###    Multiple Operations    ###
        
    elif len(split_equation) == 5:
        if split_equation[1] == "plus":
            first_op = (int(split_equation[0]) + int(split_equation[2]))
        elif split_equation[1] == "minus":
            first_op = (int(split_equation[0]) - int(split_equation[2]))
        elif split_equation[1] == "multiplied":
            first_op = (int(split_equation[0]) * int(split_equation[2]))
        elif split_equation[1] == "divided":
            first_op = (int(split_equation[0]) / int(split_equation[2]))
        else:
            raise ValueError("unknown operation")
        if split_equation[3] == "plus":
            return (first_op + int(split_equation[4]))
        elif split_equation[3] == "minus":
            return (first_op - int(split_equation[4]))
        elif split_equation[3] == "multiplied":
            return (first_op * int(split_equation[4]))
        elif split_equation[3] == "divided":
            return (first_op / int(split_equation[4]))
        else:
            raise ValueError("unknown operation")

###    Errors    ###
        
    elif len(split_equation) == 0 or len(split_equation) == 2 or  len(split_equation) == 4:
        raise ValueError("syntax error")
 
    else:
        raise ValueError("unknown operation")
